packages/kitsu-core/src/linkRelationships/index.js

```diff
@@ -57,14 +59,21 @@ function linkArray (data, included, key, previouslyLinked) {
  * @param {Object[]} included The response included object
  * @param {string} key Name of the relationship item
  * @param {Object} previouslyLinked A mapping of already visited resources
+ * @param {Object} relationshipCache A cache object for relationship meta and links
  * @private
  */
-function linkObject (data, included, key, previouslyLinked) {
+function linkObject (data, included, key, previouslyLinked, relationshipCache) {
   data[key] = {}
   const resource = data.relationships[key].data
   const cache = previouslyLinked[`${resource.type}#${resource.id}`]
-  data[key].data = cache || link(resource, included, previouslyLinked)
-  if (data.relationships[key].links) data[key].links = data.relationships[key].links
+  data[key].data = cache || link(resource, included, previouslyLinked, relationshipCache)
+
+  const relationships = relationshipCache[`${data.type}#${data.id}#${key}`] || data.relationships[key]
+  if (!relationshipCache[`${data.type}#${data.id}#${key}`]) relationshipCache[`${data.type}#${data.id}#${key}`] = relationships
+
+  if (relationships && relationships.links) data[key].links = relationships.links
```

Comment:
Can be simplified with optional chaining

```suggestion
  if (relationships?.links) data[key].links = relationships.links
```